### PR TITLE
Update Java SDK Doc

### PR DESCRIPTION
**Pull request by weihao-statsig**
few updates

### CODE REVIEW

File: docs/server/java/_getFeatureGate.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {label: 'Kotlin', value: 'kotlin'},
+  ]}>
+  <TabItem value="java">
+
+```java
+
+APIFeatureGate featureGate = Statsig.getFeatureGate(user, "a_gate");
+System.out.println("Feature Gate: " + featureGate.getName() + ", Is Enabled?: " + featureGate.getValue());
+
+String reason = featureGate.getReason() != null ? featureGate.getReason().toString() : "No specific reason";
```
Comment (tore-statsig) [inline docs/server/java/_getFeatureGate.mdx:38]: LOL @ no specific reason

File: docs/server/java/_reference.mdx [hunk 2]
```diff
@@ -51,7 +51,9 @@ class StatsigOptions(
     var localMode: Boolean = false,
     var rulesetsSyncIntervalMs: Long = CONFIG_SYNC_INTERVAL_MS,
     var idListsSyncIntervalMs: Long = ID_LISTS_SYNC_INTERVAL_MS,
-    var dataStore: IDataStore? = null
+    var dataStore: IDataStore? = null,
+    var customLogger: LoggerInterface = defaultLogger,
+    var disableAllLogging: Boolean = false,
```
Comment (tore-statsig) [inline docs/server/java/_reference.mdx:56]: please add the loggerinterface type

File: docs/server/java/_reference.mdx [hunk 3]
```diff
@@ -51,7 +51,9 @@ class StatsigOptions(
     var localMode: Boolean = false,
     var rulesetsSyncIntervalMs: Long = CONFIG_SYNC_INTERVAL_MS,
     var idListsSyncIntervalMs: Long = ID_LISTS_SYNC_INTERVAL_MS,
-    var dataStore: IDataStore? = null
+    var dataStore: IDataStore? = null,
```
Comment (tore-statsig) [inline docs/server/java/_reference.mdx:54]: do we have the data store type as well?  might need to add that

File: docs/server/java/_getFeatureGate.mdx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# `APIFeatureGate` Data Class Overview
+
+The `APIFeatureGate` data class encapsulates the evaluation result of a feature gate for a specific user. It provides detailed information regarding whether the feature gate is enabled or disabled etc.
```
Comment (tore-statsig) [inline docs/server/java/_getFeatureGate.mdx:6]: The enabled/disabled is no different than `checkGate` - its mostly about getting metadata about the gate

**Comment by jasonwzm** [comment]
🚀 Deployed on https://6610652d914aad4d9f89c9b1--cozy-fox-0defec.netlify.app

**Review by tore-statsig — APPROVED** [review]

File: docs/server/java/_reference.mdx (hunk 3 above)
Comment (weihao-statsig) [reply docs/server/java/_reference.mdx:54]: localDataFile havent release yet, so this is fine now. We can update next week

File: docs/server/java/_reference.mdx (hunk 2 above)
Comment (weihao-statsig) [reply docs/server/java/_reference.mdx:56]: added

**Pull request merged**